webpay/pay/views.py

```diff
@@ -92,11 +92,14 @@ def process_pay_req(request, data=None):
 
     # Assert pricePoint is valid.
     try:
-        marketplace.get_price(pay_req['request']['pricePoint'])
+        price_data = marketplace.get_price(pay_req['request']['pricePoint'])
     except UnknownPricePoint:
         log.exception('UnknownPricePoint calling get price_price()')
         return app_error(request, code=msg.BAD_PRICE_POINT)
 
+    # Check price and see if it's a free product or not.
+    request.session['payment_required'] = price_data.get('price') != '0.00'
```

Comment:
Is this easier than returning the price and doing the 0.00 calculation on the front end? Not sure which is easier, but just that one data point contains more information than the other.